internal/worker/trace/client.go

```diff
@@ -169,3 +187,59 @@ func newResource(serviceName, serviceID string) *resource.Resource {
 		semconv.ServiceInstanceID(serviceID),
 	)
 }
+
+type tailSamplingProcessor struct {
+	bsp sdktrace.SpanProcessor
+
+	logger logger.Logger
+
+	threshold time.Duration
+}
+
+// OnStart is called when a span is started. It is called synchronously and
+// should not block.
+func (p *tailSamplingProcessor) OnStart(parent context.Context, s sdktrace.ReadWriteSpan) {
+	p.bsp.OnStart(parent, s)
+}
+
+// OnEnd is called when span is finished. It is called synchronously and hence
+// should not block.
+func (p *tailSamplingProcessor) OnEnd(s sdktrace.ReadOnlySpan) {
+	// If the span has an error status, we want to export it regardless of the
+	// sampling rate.
+	if status := s.Status(); status.Code == codes.Error {
+		p.bsp.OnEnd(s)
+		return
+	}
+
+	// Span duration is the time it took for the span to complete.
+	spanDuration := s.EndTime().Sub(s.StartTime())
+	if spanDuration >= p.threshold {
+		p.bsp.OnEnd(s)
+		return
+	}
+
+	// If the span duration is less than the threshold, we want to drop it.
```

Comment:
OK, so setting a threshold of `0s` is how you sample everything. I guess we might at some point want more configuration options to control the batch span processor to make it bigger or block when the queue is full.

Reply:
100%, but we need to start somewhere.